packages/e2e-test-utils/src/insert-block.js

```diff
@@ -7,12 +12,24 @@ import { searchForBlock } from './search-for-block';
  * Opens the inserter, searches for the given term, then selects the first
  * result that appears.
  *
- * @param {string} searchTerm The text to search the inserter for.
+ * @param {string} searchTerm  The text to search the inserter for.
  */
 export async function insertBlock( searchTerm ) {
 	await searchForBlock( searchTerm );
 	const insertButton = (
 		await page.$x( `//button//span[contains(text(), '${ searchTerm }')]` )
 	 )[ 0 ];
 	await insertButton.click();
+
+	// Wait for the block to show up for 500ms, but do not fail the test if it
+	// doesn't exist. To find the block, we match elements with a data-type whose
+	// value ends with the search term in kebab case. For example, if "Container
+	// without paragraph" was passed as the search term, it matches a block with
+	// `data-type="test/container-without-paragraph"`.
+	try {
+		await page.waitForSelector(
+			`.wp-block[data-type$="${ kebabCase( searchTerm ) }"]`,
+			{ timeout: 500 }
```

Comment:
> Wait for the block to show up for 500ms, but do not fail the test if it  doesn't exist.

Why do we need a timeout, and why don't  we want to fail if the block is n ot existant?

Reply:
The search term isn't guaranteed to exactly match the end of the block name, so I wasn't convinced we could rely only on this selector. Since this is only a test helper, I didn't want it to block any existing tests should the selector fail to find the block for that reason. That is also why I added a timeout.

Reply:
Well, in that case, I should ask why we need this addition at all 🤷 if we don't guarantee it 100% (basically it's just a hidden `waitFor( 500 )` ) 

IMO, we should one of these two options:

 1- Find a consistent way to add this check without timeout
 2- Check why  the "click" is "async" and why we need to wait for anything for the block to show up? I can't think of where we'd be using async behavior here.

Reply:
> basically it's just a hidden `waitFor( 500 )`

yes, correct, because that makes the tests pass

> Find a consistent way to add this check without timeout

I can try removing the timeout and allowing it to fail to see what happens. 

> why we need to wait for anything for the block to show up

I've exhausted my ideas on this front and didn't find anything :/

Reply:
As I suspected, many of the existing tests don't pass the exact block name, so the selector fails. Which makes sense, this component is meant to search for x and insert the first result, regardless of whether it's a match.

Reply:
There might other  ways to perform the check (count blocks before and after)

Reply:
that's a decent idea, will try

Reply:
It seems that it's not perfect either since somtimes inserting a block inserts more than one (Cover)

Reply:
maybe if we just check that the block list length has increased

Reply:
actually if we use `getAllBlocks` we can just check that the reference has changed, I think

Reply:
Yep, that seems to do the trick!